src/libs/tools/src/plugindatabase.cpp

```diff
@@ -379,6 +380,108 @@ std::vector<PluginSpec> ModulesPluginDatabase::lookupAllProvides (std::string co
 }
 
 
+class PluginVariantsDatabase::Impl
+{
+public:
+	Impl ()
+	{
+	}
+	~Impl ()
+	{
+	}
+	Modules modules;
```

Comment:
We already have Modules within  ModulesPluginDatabase we inherit from?

Make impl protected or make them friends to access impl (should not affect ABI).